cmd/ocm/plugin/list/cmd.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright © 2020 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/openshift-online/ocm-cli/pkg/table"
+	"github.com/spf13/cobra"
+)
+
+// Cmd represents the plugin command
+var Cmd = &cobra.Command{
+	Use:   "list",
+	Short: "list ocm plugins",
+	Long:  "list all the plugins under the user executable path",
+	RunE:  run,
+}
+
+var args struct {
+	nameOnly bool
+}
+
+func init() {
+	flags := Cmd.Flags()
+	flags.BoolVar(
+		&args.nameOnly,
+		"nameonly",
+		false,
+		"Show the plugin name only",
+	)
+}
+
+func run(cmd *cobra.Command, argv []string) error {
+	defaultPath := filepath.SplitList(os.Getenv("PATH"))
+	newPath := uniquePath(defaultPath)
+	pluginPrefix := "ocm-"
+
+	columns := "NAME,PATH"
+	paddingByColumn := []int{20, 30}
+
+	table.PrintPadded(os.Stdout, strings.Split(columns, ","), paddingByColumn)
+
+	for _, dir := range newPath {
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
```

Comment:
+1 to continuing after non-existant PATH entries, with or without warning.  
Most programs, including `which` ignore them silently.

$PATH may contain empty entries which mean current working dir, so it's better to first do:

```
if dir == "" {
  dir = "."
}
```

> A zero-length prefix is a legacy feature that indicates the current working directory. It appears as two adjacent <colon> characters ( "::" ), as an initial <colon> preceding the rest of the list, or as a trailing <colon> following the rest of the list. A strictly conforming application shall use an actual pathname (such as .) to represent the current working directory in PATH.  
-- [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08_03)